### PR TITLE
[fi] Add templates for EM/LFI & clk glitching

### LINKED ISSUE #250: [fi] LFI template
To help others implementing device support for their LFI setup, a new LFI dummy class shall be added to the fi scripts.

### LINKED ISSUE #251: [fi] EMFI template
To help others implementing device support for their EMFI setup, a new EMFI dummy class shall be added to the fi scripts.

### LINKED ISSUE #252: [fi] Clock glitching template
To help others implementing device support for their clock glitching setup, a new clock glitching dummy class shall be added to the fi scripts.

### PR DESCRIPTION
**Pull request by nasahlpa**
This PR adds dummy drivers for EMFI, LFI, and CLK glitching FI gear that can be used by others as a template for new FI gear.

Closes #250
Closes #251
Closes #252

### CODE REVIEW

File: fault_injection/fi_gear/dummy/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+All drivers need to implement these functions:
+- init:
+    - Initializes the FI gear.
+- generate_fi_parameters:
+    - Generate parameters for the FI parameter sweep. This can happen either
+      randomly or deterministically.
+- arm_trigger:
+    - Configure the FI gear with the current FI parameters and arm the trigger.
+- reset:
+    - After a power cycle of the DUT, some FI gear also need to reset it.
```
Comment (vogelpi) [inline fault_injection/fi_gear/dummy/README.md:14]: I am not sure I understand this. Do you mean the FI gear needs to reset the DUT?

File: fault_injection/fi_gear/dummy/dummy_emfi.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+from fi_gear.utility import random_float_range
+
+""" Template for EMFI gear.
+Acts as a template for electromagnetic FI gear.
```
Comment (vogelpi) [inline fault_injection/fi_gear/dummy/dummy_emfi.py:8]: Nit: add " with XY table".

File: fault_injection/fi_gear/dummy/dummy_lfi.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+from fi_gear.utility import random_float_range
+
+""" Template for LFI gear.
+Acts as a template for laser FI gear.
```
Comment (vogelpi) [inline fault_injection/fi_gear/dummy/dummy_lfi.py:8]: Nit: add " with XY table".

File: fault_injection/fi_gear/dummy/dummy_emfi.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+        Args:
+            A dict containing the FI parameters.
+        """
+        print(f"Arming DummyEMFI trigger with x_position="
+              f"{fault_parameters['x_position']} y_position="
+              f"{fault_parameters['y_position']}, voltage="
+              f"{fault_parameters['voltage']}, pulse_width="
+              f"{fault_parameters['pulse_width']}, and trigger_delay="
+              f"{fault_parameters['trigger_delay']}")
```
Comment (vogelpi) [inline fault_injection/fi_gear/dummy/dummy_emfi.py:47]: Nit: the formatting here is a bit unfortunate as the string and the corresponding variable reference are not on the same line. Maybe the initial string can be split up into two substrings such that things match?

File: fault_injection/fi_gear/dummy/dummy_emfi.py [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+                                                      self.y_position_step)
+        parameters["voltage"] = random_float_range(self.voltage_min,
+                                                   self.voltage_max,
+                                                   self.voltage_step)
+        parameters["pulse_width"] = random_float_range(self.pulse_width_min,
+                                                       self.pulse_width_max,
+                                                       self.pulse_width_step)
+        parameters["trigger_delay"] = random_float_range(self.trigger_delay_min,
+                                                         self.trigger_delay_max,
+                                                         self.trigger_step)
```
Comment (vogelpi) [inline fault_injection/fi_gear/dummy/dummy_emfi.py:70]: Do you also want to add an example for a deterministic XY sweep?

**Review by vogelpi — APPROVED** [review]
Thanks @nasahlpa , this looks nice!

File: fault_injection/fi_gear/dummy/README.md (hunk 1 above)
Comment (nasahlpa) [reply fault_injection/fi_gear/dummy/README.md:14]: No, the FI gear itself (i.e., Husky needs a re-initialization) needs a reset after resetting/reprogramming OpenTitan/FPGA. Other FI gear (LFI, ...) does not need to implement this.

Comment (vogelpi) [reply fault_injection/fi_gear/dummy/README.md:14]: Ah ok, thanks for explaining. So the sentence should probably read, `...some FI gear also needs to be reset.`

File: fault_injection/fi_gear/dummy/dummy_emfi.py (hunk 5 above)
Comment (nasahlpa) [reply fault_injection/fi_gear/dummy/dummy_emfi.py]: I've added an example for a deterministic XY sweep to LFI & EMFI.

**Pull request merged**